man/sd_bus_error.xml

```diff
@@ -138,234 +150,188 @@
-    not copied internally, and must remain valid for the lifetime of
-    <parameter>e</parameter>.</para>
-
-    <para><function>sd_bus_error_set_errno</function> will set
-    <structfield>name</structfield> based on an errno-like value.
-    <citerefentry project='die-net'><refentrytitle>strerror</refentrytitle><manvolnum>3</manvolnum></citerefentry>
+    <function>return</function> statement. If
+    <parameter>message</parameter> is NULL no message is set. This
+    call can fail if no memory may be allocated for the name and
+    message strings, in which case an <constant>SD_BUS_ERROR_NO_MEMORY</constant> error might be set
+    instead and -ENOMEM returned.</para>
+
+    <para><function>sd_bus_error_setf()</function> is similar to
+    <function>sd_bus_error_set()</function>, but takes a <citerefentry
+    project='man-pages'><refentrytitle>printf</refentrytitle><manvolnum>3</manvolnum></citerefentry>
+    format string and corresponding arguments to generate the
+    <structfield>message</structfield> field.</para>
+
+    <para><function>sd_bus_error_set_const()</function> is similar to
+    <function>sd_bus_error_set()</function>, but the string parameters
+    are not copied internally, and must hence remain constant and
+    valid for the lifetime of <parameter>e</parameter>. Use this call
+    to avoid memory allocations when setting error structures. Since
+    this call does not allocate memory it will not fail with an
+    out-of-memory condition, as
+    <function>sd_bus_error_set()</function> can, as described
+    above. Alternatively, the
+    <constant>SD_BUS_ERROR_MAKE_CONST()</constant> macro may be used
+    to generate a literal, constant bus error structure
+    on-the-fly.</para>
+
+    <para><function>sd_bus_error_set_errno()</function> will set
+    <structfield>name</structfield> from an
+    <varname>errno</varname>-like value that is converted to a D-Bus
+    error. <citerefentry
+    project='die-net'><refentrytitle>strerror_r</refentrytitle><manvolnum>3</manvolnum></citerefentry>
     will be used to set <structfield>message</structfield>. Well-known
     D-Bus error names will be used for <structfield>name</structfield>
-    if available, otherwise a name in the
-    <literal>System.Error</literal> namespace will be generated.
-    </para>
-
-    <para><function>sd_bus_error_set_errnof</function> is similar to
-    <function>sd_bus_error_set_errno</function>, but in addition to
-    <parameter>name</parameter>, takes a
-    <citerefentry project='man-pages'><refentrytitle>printf</refentrytitle><manvolnum>3</manvolnum></citerefentry>
-    format and corresponding arguments.
-    <structfield>name</structfield> will be generated from
+    if applicable, otherwise a name in the
+    <literal>System.Error.</literal> namespace will be generated. The
+    error number specified may either be passed in negative or
+    positive, the call will implicitly take the absolute value. The
```

Comment:
Sounds awkward. Maybe: "The sign of the specified error number is ignored. Absolute value is used implicitly."